### PR TITLE
HPCC-13941 Make unicode conversions thread safe.

### PR DESCRIPTION
**Pull request by jakesmith**
Unicode page conversions of the same code page, used the same
converter instance and were thread unsafe.
Use a codepage table per thread (using a thread variable).

Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@richardkchapman - please review

File: rtl/eclrtl/eclrtl.cpp [hunk 1]
```diff
@@ -307,23 +307,48 @@ class RTLUnicodeConverter : public CInterface
...
+        {
+            /* Store them in a global array for eventual disposal
+             * NB: in C++11, we would use TLS thread_local for the maps and would do away with the global map
+             */
+            CriticalBlock b(ucmCrit);
+            prevThreadTerminator = addThreadTermFunc(clearUnicodeConverterMap);
+        }
+        // Use thread terminator hook to clear them up on thread exit.
+        // NB: May need to revisit if not on a jlib Thread.
+        unicodeCoverterThreadMaps->append(unicodeConverterMap);
```
Comment (richardkchapman) [inline rtl/eclrtl/eclrtl.cpp]: Seems to me that this append needs protection from the critsec, but the addThreadTermFunc call above does not.

I'm not sure what the benefit of the unicodeConverterThreadMaps is really - by the time it is used to cleanup /release the process is closing anyway.

Comment (jakesmith) [inline rtl/eclrtl/eclrtl.cpp]: You're right and that's the way it was before a final edit that added the block braces.. - whoops.

**Comment by jakesmith** [comment]
@richardkchapman - corrected mistake you spotted

**Pull request merged**